spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,234 @@
+require 'rails_helper'
+
+RSpec.describe ArticlesController, :type => :controller do
+  let(:valid_credentials) { ActionController::HttpAuthentication::Basic.encode_credentials("dhh", "secret") }
+
+  describe 'GET #index' do
+    let!(:articles) { create_list(:article, 3) }
+
+    before { get :index }
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'assigns articles' do
+      expect(assigns(:articles)).to eq(articles)
+    end
+
+    it "renders 'index' template" do
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe 'GET #show' do
+    let(:article) { create(:article) }
+
+    before { get :show, params: { id: article.id } }
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'assigns article' do
+      expect(assigns(:article)).to eq(article)
+    end
+
+    it "renders 'show' template" do
+      expect(response).to render_template(:show)
+    end
```

Comment:
What about behaves as credentials checker here?

Reply:
I have this line in articles_controller

`http_basic_authenticate_with name: "dhh", password: "secret", except: [:index, :show]`

so I think I don't really need to test it